### ISSUE TITLE
Object passing through

### ISSUE THREAD

**Issue by impguard** [liabru/matter-js#672]
So this problem may be related to CCD in #5, but I don't believe so.

We're making a simple pong style game except that instead of just a paddle, you have a flipper and a paddle (think pong paddle and pinball flippers). Our balls are flying around at pretty reasonable speeds. It's not a bullet so I don't believe it's "teleporting" through our objects. (Visually, I can see it "pass through" my objects, though I am using the built in Renderer so I don't know if it's doing any smoothing.

I've made the paddles and flippers pretty large to test as well, and the ball seems to be passing through them.

Not all the time, mind you. I've been debugging by playing around with my settings, and it almost feels like the ball passes through "between" my paddles and flippers. If I remove flippers and leave just a paddle, no pass through seems to happen. As soon as I introduce both paddles and flippers, the ball seems to pass through between them.

I've tried squashing both paddles and flippers on top of each other, so that they resemble three rectangles that intersect. The ball seems to still randomly pass through them.

Some additional notes that may play an impact:
- We are clamping the ball velocity, so to test, I'm clamping it so that the velocity is set every tick to a constant value.
- There's no gravity at all.

Not looking for concrete answers since there's probably a lot of variables at play here, but I wanted to see if anyone had any idea of what to look into?

**Comment by liabru** [comment]
It does sound like a CCD issue, for something like this I would highly suggest using a lower delta and multiple `Engine.update` per step (e.g. call 3 times per frame using a delta of `(1/3)/60 * 1000`ms). Also you should make your flippers as thick as possible but render them as if they were thin, using collision filters so they don't jam against walls.

There's actually an awesome [pinball example](https://codepen.io/lonekorean/pen/KXLrVX/) by @lonekorean you should check out and he also wrote a great [post about it](https://codersblock.com/blog/javascript-physics-with-matter-js/) which includes how he solved the issue you're having for now until CCD is sorted.

**Comment by impguard** [comment]
I'll take a closer look at that post, though I have referenced it before. I'll also see if I can get a quick video to demonstrate what I'm talking about.

In short, however, to test my theories I have tried making the paddles enormous (like 10x) the size of my ball, but the ball still seems to slip "in between" the gap

**Comment by impguard** [comment]
Actually, I have a question that may make this clearer for me. Say an object is moving fast, but not fast enough to fully "teleport" through another object. However, it is moving fast enough that it moves from a frame of absolutely no collision to a frame of being 100% enclosed by the other object.

In this case, would the object continue penetrating or be "ejected"?

If it's the former, then, as far as I understand, as long as a bullet is moving fast enough to completely penetrate another object (not teleport through), then it still breaks the collision model?

**Comment by juancaacuna** [comment]
@impguard Where you able to fix the problem? I'm having the same issue.

**Comment by impguard** [comment]
@juancaacuna No, the project paused, but the issue seemed to continue to happen. Could be related to CCD, but definitely felt strange since the projectile was moving pretty slowly and only seemed to intersect between items.
